src/main/java/gov/nasa/jpf/constraints/expressions/PropositionalCompound.java

```diff
@@ -93,10 +93,19 @@ public LogicalOperator getOperator() {
 
   @Override
   public void print(Appendable a, int flags) throws IOException {
+    if(left == null){
+      System.out.println(operator.toString());
+      right.print(System.out);
+      System.out.flush();
+    }
     a.append('(');
     left.print(a, flags);
     a.append(' ').append(operator.toString()).append(' ');
-    right.print(a, flags);
```

Comment:
Why can right be null? I don't think it is a good idea to just go ahead and print 'null' if this is a malformed expression. This should yield an exception